appinventor/appengine/src/com/google/appinventor/client/widgets/DropDownButton.java

```diff
@@ -246,6 +246,20 @@ public void replaceLastItem(DropDownItem item) {
     items.remove(items.size()-1);
     items.add(menu.addItem(item.caption, true, item.command));
   }
+  
+  public void replaceSpecificItem(DropDownItem item, int num) {
```

Comment:
Rather than change the Text and action of a menu item, we'd prefer to create both menu items and then make them enabled/visible as appropriate.

Reply:
@SusanRatiLane Sorry for late replies, I was busy with my GSoC Proposal.
So will it be okay to add the ``Delete From Trash`` button to the ``My Projects`` file DropDown?

Reply:
@beinganonymous Yes. Just make it invisible and disabled when not in use.